File: atividade java/Media.java
import java.util.Scanner;

public class Media {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite a primeira nota: ");
        double nota1 = scanner.nextDouble();

        System.out.print("Digite o peso da primeira nota: ");
        double peso1 = scanner.nextDouble();

        System.out.print("Digite a segunda nota: ");
        double nota2 = scanner.nextDouble();

        System.out.print("Digite o peso da segunda nota: ");
        double peso2 = scanner.nextDouble();

        System.out.print("Digite a terceira nota: ");
        double nota3 = scanner.nextDouble();

        System.out.print("Digite o peso da terceira nota: ");
        double peso3 = scanner.nextDouble();

        double mediaPonderada = calcularMediaPonderada(nota1, peso1, nota2, peso2, nota3, peso3);

        System.out.println("A média ponderada das notas é: " + mediaPonderada);

        scanner.close();
    }

    private static double calcularMediaPonderada(double nota1, double peso1, double nota2, double peso2, double nota3, double peso3) {
        double somaPesos = peso1 + peso2 + peso3;
        double mediaPonderada = (nota1 * peso1 + nota2 * peso2 + nota3 * peso3) / somaPesos;
        return mediaPonderada;
    }
}

File: atividade haskell/media.hs
main :: IO ()
main = do
    putStrLn "Digite a primeira nota:"
    inputNota1 <- getLine
    let nota1 = read inputNota1 :: Double

    putStrLn "Digite o peso:"
    inputPeso1 <- getLine
    let peso1 = read inputPeso1 :: Double

    putStrLn "Digite a segunda nota:"
    inputNota2 <- getLine
    let nota2 = read inputNota2 :: Double

    putStrLn "Digite o peso:"
    inputPeso2 <- getLine
    let peso2 = read inputPeso2 :: Double

    putStrLn "Digite a terceira nota:"
    inputNota3 <- getLine
    let nota3 = read inputNota3 :: Double

    putStrLn "Digite o peso:"
    inputPeso3 <- getLine
    let peso3 = read inputPeso3 :: Double

    let mediaPonderada = calcularMediaPonderada nota1 peso1 nota2 peso2 nota3 peso3

    putStrLn $ "Média ponderada: " ++ show mediaPonderada

calcularMediaPonderada :: Double -> Double -> Double -> Double -> Double -> Double -> Double
calcularMediaPonderada nota1 peso1 nota2 peso2 nota3 peso3 =
    (nota1 * peso1 + nota2 * peso2 + nota3 * peso3) / (peso1 + peso2 + peso3)